mesonbuild/utils/universal.py

```diff
@@ -756,40 +757,54 @@ def windows_detect_native_arch() -> str:
             raise EnvironmentException('Unable to detect native OS architecture')
     return arch
 
-def detect_vcs(source_dir: T.Union[str, Path]) -> T.Optional[T.Dict[str, str]]:
+@dataclasses.dataclass
+class VcsData:
+    name: str
+    cmd: str
+    repo_dir: str
+    get_rev: 'T.List[str]'
+    rev_regex: str
```

Comment:
Since there's only one case where this isn't `'(.*)`, I think it makes sense to give this a default value of `(.*)` and then not having to repeat ourselves later.

Reply:
If every case had the same value, sure. But in this case I'd say explicit is better than implicit.